### ISSUE TITLE
Cannot create C# function: Request to cli-feed-v4.json failed, unable to get local issuer certificate

### ISSUE THREAD

**Issue by smayesdxc** [microsoft/vscode-azurefunctions#3667]
### Description of issue
When attempting to create a new C# Azure Function using the extension in VS Code, I receive the following error.

```
request to https://functionscdn.azureedge.net/public/cli-feed-v4.json failed, reason: unable to get local issuer certificate
```

Note that I am running this on a corporate-issued laptop, which has security software Zscaler on it. Zscaler has its own root CA certificate which sometimes interferes with third-party software such as node, npm, etc. However, I can usually add the certificate to software such as npm to get it to work just fine with SSL connections. I am not sure but am wondering if perhaps this is causing the problem.

### Version of VS Code
Version: 1.78.0 (user setup)
Commit: 252e5463d60e63238250799aef7375787f68b4ee
Date: 2023-05-03T20:09:00.748Z
Electron: 22.4.8
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No

### Operating system
Windows 10 Enterprise, Version 21H2 (OS Build 19044.2846)

### Version of the Azure Functions VS Code extension
v1.10.5

### Reproducible steps that cause the issue

1. Open a new window of VS Code
2. Navigate to the Azure tab on the left
3. Under **Workspace**, click the **Create function...** button
4. Select a folder from the local filesystem
5. When prompted with **Select a language**, select **C#**
6. When prompted with **Select a runtime**, an empty box appears briefly, then disappears with the following error message displayed in the lower-right of the window.

```
request to https://functionscdn.azureedge.net/public/cli-feed-v4.json failed, reason: unable to get local issuer certificate
```

### What was expected to see, versus what was actually seen
I expected to see a list of valid runtimes to select from.

### Images, animations, or a link to a video showing the issue occurring
![AzFuncVsCodeExtIssue](https://user-images.githubusercontent.com/12766851/236846603-b838bae0-92fc-49b0-b5f9-7ef1831d0932.gif)

### Errors from the Dev Tools Console

```
log.ts:431  WARN Via 'product.json#extensionEnabledApiProposals' extension 'ms-vscode.vscode-selfhost-test-provider' wants API proposal 'testContinuousRun' but that proposal DOES NOT EXIST. Likely, the proposal has been finalized (check 'vscode.d.ts') or was abandoned.
log.ts:441   ERR Extension 'github.vscode-pull-request-github' wants API proposal 'contribEditorLineNumberMenu' but that proposal DOES NOT EXIST. Likely, the proposal has been finalized (check 'vscode.d.ts') or was abandoned.
log.ts:431  WARN [redhat.vscode-xml]: Cannot register 'redhat.telemetry.enabled'. This property is already registered.
log.ts:441   ERR [Extension Host] (node:17748) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `Code --trace-deprecation ...` to show where the warning was created)
console.ts:137 [Extension Host] (node:17748) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `Code --trace-deprecation ...` to show where the warning was created)
I @ console.ts:137
log.ts:421  INFO [perf] Render performance baseline is 38ms
notificationsAlerts.ts:42 request to https://functionscdn.azureedge.net/public/cli-feed-v4.json failed, reason: unable to get local issuer certificate
c @ notificationsAlerts.ts:42
notificationsAlerts.ts:42 request to https://functionscdn.azureedge.net/public/cli-feed-v4.json failed, reason: unable to get local issuer certificate
c @ notificationsAlerts.ts:42
log.ts:431  WARN Settings pattern "application.*" doesn't match any settings
log.ts:431  WARN SettingsEditor2: Settings not included in settingsLayout.ts: interactiveEditor.editMode, accessibility.verbosity.diff-editor, accessibility.verbosity.terminal
notificationsAlerts.ts:42 request to https://functionscdn.azureedge.net/public/cli-feed-v4.json failed, reason: unable to get local issuer certificate
c @ notificationsAlerts.ts:42
notificationsAlerts.ts:42 request to https://functionscdn.azureedge.net/public/cli-feed-v4.json failed, reason: unable to get local issuer certificate
c @ notificationsAlerts.ts:42
```

**Comment by Maheshdev-Learning** [comment]
Yeah I'm also facing the same issue with new version of VS code update 1.78....

**Comment by alexweininger** [comment]
A change released in the VS Code 1.78 April release breaks loading OS certificates on Windows. Meaning self-signed certificates and CA certificates picked up from a network proxy are currently broken. See https://github.com/microsoft/vscode/issues/181404 for more details.

It looks like VS Code is planning to include a fix in the [1.78.2 release (April recovery 2)](https://github.com/microsoft/vscode/issues/181866).

Until that is released, [other users](https://github.com/microsoft/vscode-azure-account/issues/792#issuecomment-1540127373) have had success installing the [win-ca extension](https://marketplace.visualstudio.com/items?itemName=ukoloff.win-ca), which makes Trusted Root Certificates on Windows available to VS Code extensions.

Another workaround if your CA certificate is provided by a network proxy is to configure the proxy related settings in VS Code.

**Comment by alexweininger** [comment]
VS Code just released version 1.78.2 which should fix the certificates issue, could you please update VS Code and retry the operation?
